types/jsonfile/index.d.ts

```diff
@@ -49,12 +49,12 @@ export type JFWriteOptions = string | {
 
 export type JFReadCallback = (err: NodeJS.ErrnoException | null, data: any) => void;
 
-export function readFile(file: Path, options?: JFReadOptions, callback?: JFReadCallback): void;
-export function readFile(file: Path, callback: JFReadCallback): void;
+export function readFile(file: Path, options?: JFReadOptions, callback?: JFReadCallback): Promise<any>;
+export function readFile(file: Path, callback: JFReadCallback): Promise<any>;
 
 export function readFileSync(file: Path, options?: JFReadOptions): any;
 
-export function writeFile(file: Path, obj: any, options?: JFWriteOptions, callback?: WriteCallback): void;
-export function writeFile(file: Path, obj: any, callback: WriteCallback): void;
+export function writeFile(file: Path, obj: any, options?: JFWriteOptions, callback?: WriteCallback): Promise<any>;
```

Comment:
From reading the package's readme, it seems like these functions only return promises when the callback is not provided. So this should be a separate overload, not an actual change to the existing return type.